engines/query-sparql/test/util.ts

```diff
@@ -1,42 +1,26 @@
-/* eslint-disable jest/require-top-level-describe */
-import { resolve } from 'node:path';
-import { Polly } from '@pollyjs/core';
-import { setupPolly } from 'setup-polly-jest';
+import * as fs from 'node:fs';
+import * as path from 'node:path';
 
-const NodeHttpAdapter = require('@pollyjs/adapter-node-http');
-const FSPersister = require('@pollyjs/persister-fs');
+const md5 = require('md5');
 
-const recordingsDir = resolve(__dirname, './assets/http');
+const fetchFn = globalThis.fetch;
 
-Polly.register(FSPersister);
-Polly.register(NodeHttpAdapter);
-
-// Configure everything related to PollyJS
-export function usePolly() {
-  const pollyContext = mockHttp();
-
-  beforeEach(() => {
-    pollyContext.polly.server.any().on('beforePersist', (req, recording) => {
-      recording.request.headers = recording.request.headers.filter(({ name }: any) => name !== 'user-agent');
-    });
-  });
-
-  afterEach(async() => {
-    await pollyContext.polly.flush();
-  });
-}
-
-// Mocks HTTP requests using Polly.JS
-export function mockHttp() {
-  return setupPolly({
-    adapters: [ NodeHttpAdapter ],
-    persister: FSPersister,
-    persisterOptions: { fs: { recordingsDir }},
-    recordFailedRequests: true,
-    matchRequestsBy: {
-      headers: {
-        exclude: [ 'user-agent' ],
-      },
-    },
-  });
-}
+export async function fetch(...args: Parameters<typeof fetchFn>): ReturnType<typeof fetchFn> {
```

Comment:
I would prefer to stick with Polly if possible.
I'm afraid we'll introduce other problems with our custom script that Polly has had years to stabilize over.

If the problem is related to long filenames, then I suspect Polly will have a way to somehow shortening them.

Reply:
> I had a go at extending this persistor to make my own custom persistor but it was non-trivial. I will briefly continue with that attempt.

In starting to look into this I've noticed that Polly is not caching all responses needed in e2e tests [and 23 tests fail when tests are run offline](https://pastebin.com/K9N8Yw75). This means that Comunica tests will start failing if any of fragments.dbpedia.org, xmlns.com or www.rubensworks.net are down and to me serves as a good argument for updating this mechanism anyway.